frontend/src/pages/Admin/index.tsx

```diff
@@ -1,9 +1,32 @@
-import React from "react";
+import React, { FC, useEffect } from "react";
 import { Outlet, useParams } from "react-router-dom";
+import apiAxios from "../../utils/apiAxios";
+import { useKeycloak } from "@react-keycloak/web";
 import Sidebar from "../../components/Sidebar";
 import Main from "../../components/Main";
 
-export const Admin = () => {
+export const Admin: FC = () => {
+  const { initialized, keycloak } = useKeycloak();
+
+  /**
+   * Check keycloak init and auth on component load.
+   *
+   * @todo Need a better way to get access to the keycloak token.
```

Comment:
what is the problem right now that eventually needs to be bettered?

Reply:
I see that this is a little opaque yes. Originally, I felt I needed to write a really long todo to explain this but I figured out a more terse explanation. I'll include it.

Essentially, we don't want to be passing the keycloak token from the component each time we need to make an API call. But the useKeycloak hook can only be called in components and nowhere else. So, we need to look at another library or something to get access to the token - preferably from within apiAxios.tsx.

Reply:
yeah, we need to get and check the keycloak token to authenticate user login, then create our own new internal token to use between frontend and backend to manage the user's session. in other words, we should be able to replace keycloak auth at any time (say, with our own login page) and still have the rest of the code work unchanged.

Reply:
Oh really? I wasn't aware that we were planning that.

Reply:
haha, i think this might have been a comment left over from when i added the code, when it was relevant where the token was 'test-token'